### PR TITLE
[CP] LRSD-7193 - Add permission layer for spring boot CX to allow custom roles

### PR DESCRIPTION
**Pull request by abfranca**
https://liferay.atlassian.net/browse/LRSD-7193

### CODE REVIEW

**Comment by abfranca** [comment]
ci:test:sf

**Comment by abfranca** [comment]
ci:test:sf

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java [hunk 1]
```diff
@@ -235,8 +243,18 @@ HttpHeaders.AUTHORIZATION, _getCredentials()
...
 
-	private boolean _isPartner() {
-		return true;
+	private boolean _isPartner(Jwt jwt) {
+		ArrayList<String> userRoles = _userAccountService.getUserRoles(jwt);
+
+		if (userRoles.contains(RoleConstants.NAME_ADMINISTRATOR) ||
+			userRoles.contains(RoleConstants.NAME_LIFERAY_STAFF) ||
+			userRoles.contains(RoleConstants.NAME_PARTNER) ||
+			userRoles.contains(RoleConstants.NAME_PROVISIONING_ADMIN)) {
```
Comment (kylembischof) [inline workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java:426]: is there a reason we need Provisioning Admin role checked separately from Liferay Staff?

Comment (kylembischof) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java:426]: @jennyhchen Felipe mentioned that you were the one to suggest adding this role check for Provisioning Admin. Do we really need this role? Realistically, there should never be a Provisioning Admin that is not also Liferay staff, right?

Comment (kylembischof) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java:426]: @abfranca can we remove this role for now and just use Liferay Staff then. Seems like Provisioning Admin is not necessary for this use case. Thanks!

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Felipe Franca
+ */
+@Component
+public class UserAccountService extends BaseRestController {
```
Comment (amosfong) [inline workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java:26]: can you move this class to the REST controller? BChan told us to just put in the REST controller unless the service is used in multiple places

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	public ArrayList<String> getUserRoles(Jwt jwt) {
```
Comment (amosfong) [inline workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java:28]: Return a generic List object

Comment (amosfong) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java:28]: rename to getUserRoleNames

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		JSONObject userJSONObject = _getMyUserAccountJSONObject(jwt);
+
+		JSONArray roleBriefsJSONArray = userJSONObject.getJSONArray(
+			"roleBriefs");
+
+		ArrayList<String> rolesNames = new ArrayList<>();
```
Comment (amosfong) [inline workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java:34]: Use generic objects (ie list) to pass around unless you need a specific ArrayList method

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java [hunk 5]
```diff
@@ -235,11 +417,23 @@ HttpHeaders.AUTHORIZATION, _getCredentials()
 		return null;
 	}
 
-	private boolean _isPartner() {
-		return true;
+	private boolean _isPartner(Jwt jwt) {
```
Comment (amosfong) [inline workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java:420]: can you rename this method? it's no longer looking for just partners. Maybe hasPartnerPublishDatePermission or hasEarlyPublishAccess

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java (hunk 1 above)
Comment (abfranca) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java]: Done!

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java (hunk 2 above)
Comment (abfranca) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java]: Done!

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java (hunk 3 above)
Comment (abfranca) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java]: Done!

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java (hunk 4 above)
Comment (abfranca) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/service/UserAccountService.java]: Done!

File: workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java (hunk 5 above)
Comment (abfranca) [reply workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java]: Done!

**Comment by amosfong** [comment]
one more thing, can you remove the constants fields that aren't used? (prov admin, etc)

then once https://github.com/liferay-customer-solutions/liferay-portal/pull/394 is merged we can rebase and forward this

**Comment by abfranca** [comment]
@amosfong i've removed the unused constants and rebased with brian's master since he already merged the other pr.

**Comment by amosfong** [comment]
ci:test:sf

**Pull request closed without merging**